### PR TITLE
V1/50 search presentation

### LINKED ISSUE #50: Search for a presentation
show results from a search query
As a frontend dev,
I would like to query the api
so that I can display the users search results

**Estimation: 2h**

### PR DESCRIPTION
**Pull request by kayoumido**
As I've explained in the README, you need to import the database records to Scout, if you have records in your db, for it to work.

For you lazy people:
```shell
$ php artisan scout:import "App\Presentation"
```

### CODE REVIEW

**Review by SundayBraid — CHANGES_REQUESTED** [review]
## Error encounter

When I want to test it I encounter an error when configuring
``docker exec dev_laravel php artisan scout:import "App\Presentation"``

this get me the following exception
```
  Exception  : Please install the Algolia client: algolia/algoliasearch-client-php.

  at /var/www/vendor/laravel/scout/src/EngineManager.php:59
    55|         if (class_exists('AlgoliaSearch\Client')) {
    56|             throw new Exception('Please upgrade your Algolia client to version: ^2.2.');
    57|         }
    58| 
  > 59|         throw new Exception('Please install the Algolia client: algolia/algoliasearch-client-php.');
    60|     }
    61| 
    62|     /**
    63|      * Create a Null engine instance.

  Exception trace:

  1   Laravel\Scout\EngineManager::ensureAlgoliaClientIsInstalled()
      /var/www/vendor/laravel/scout/src/EngineManager.php:32

  2   Laravel\Scout\EngineManager::createAlgoliaDriver()
      /var/www/vendor/laravel/framework/src/Illuminate/Support/Manager.php:115

  Please use the argument -v to see more details.
``` 

And if I install the missing package (algolia) it show me this exception

```
GuzzleHttp\Exception\InvalidArgumentException  : IDN conversion failed

  at /var/www/vendor/guzzlehttp/guzzle/src/functions.php:384
    380|             if ($errors) {
    381|                 $errorMessage .= ' (errors: ' . implode(', ', $errors) . ')';
    382|             }
    383| 
  > 384|             throw new InvalidArgumentException($errorMessage);
    385|         } else {
    386|             if ($uri->getHost() !== $asciiHost) {
    387|                 // Replace URI only if the ASCII version is different
    388|                 $uri = $uri->withHost($asciiHost);

  Exception trace:

  1   GuzzleHttp\_idn_uri_convert(Object(Algolia\AlgoliaSearch\Http\Psr7\Uri))
      /var/www/vendor/guzzlehttp/guzzle/src/Client.php:220

  2   GuzzleHttp\Client::buildUri(Object(Algolia\AlgoliaSearch\Http\Psr7\Uri))
      /var/www/vendor/guzzlehttp/guzzle/src/Client.php:113

  Please use the argument -v to see more details.
```

**Review by SundayBraid — COMMENTED** [review]
The issue encountered when setup the environment is corrected now.

## Test
With postman and an authenticate user I...

- [x] Can send a request to backend with given api url and get response
- [x] Can search presentation with keyword and get an array of presentation (or empty if no result)
- [x] Can not send request with missing keywords

**Review by SundayBraid — APPROVED** [review]

File: src/app/Http/Controllers/API/PresentationController.php [hunk 1]
```diff
@@ -124,6 +124,12 @@ public function destroy(Presentation $presentation)
      */
     public function search(Request $request)
     {
-        //
+        $request->validate([
+            'keywords' => 'required'
+        ]);
+
+        return PresentationResource::collection(
+            Presentation::search($request->get('keywords'))->get()
```
Comment (fabio10000) [inline src/app/Http/Controllers/API/PresentationController.php:132]: `$request->get('keywords')` can be replaced by `$request->keywords`

File: src/app/Presentation.php [hunk 2]
```diff
@@ -2,15 +2,31 @@
...
+    {
+        return 'presentations_index';
+    }
+
+    public function toSearchableArray()
+    {
+        return [
+            'id' => $this->id,
+            'title' => $this->title,
+       ];
```
Comment (fabio10000) [inline src/app/Presentation.php:27]: should use a resource instead of array

File: src/config/scout.php [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    |
+    | Here you may configure your Algolia settings. Algolia is a cloud hosted
+    | search engine which works great with Scout out of the box. Just plug
+    | in your application ID and admin API key to get started searching.
+    |
+    */
+
+    'algolia' => [
+        'id' => env('ALGOLIA_APP_ID', ''),
+        'secret' => env('ALGOLIA_SECRET', ''),
```
Comment (fabio10000) [inline src/config/scout.php:88]: if we need to add those vars to the .env file you should include them on the .env.example file

**Review by fabio10000 — CHANGES_REQUESTED** [review]

File: src/config/scout.php (hunk 3 above)
Comment (kayoumido) [reply src/config/scout.php:88]: We don't need theses vars. We're using tntsearch. The one we need is `SCOUT_DRIVER` which i'll add to the example file

File: src/app/Http/Controllers/API/PresentationController.php (hunk 1 above)
Comment (kayoumido) [reply src/app/Http/Controllers/API/PresentationController.php:132]: okay

**Review by fabio10000 — APPROVED** [review]

**Pull request merged**